### PR TITLE
fix(prop): trusted domain

### PR DESCRIPTION
**Pull request by mnicpt**
Changes trusted domain prop to accept regex rather than strings.

### CODE REVIEW

**Review by wsbrunson — APPROVED** [review]

File: src/parent/parent.js [hunk 1]
```diff
@@ -380,7 +380,9 @@ export function parentComponent<P, X, C>({
 
       const trustedChild: boolean =
         prop && prop.trustedDomains && prop.trustedDomains.length > 0
-          ? prop.trustedDomains.includes(initialChildDomain)
+          ? prop.trustedDomains.reduce((acc, val) => {
```
Comment (gregjopa) [inline src/parent/parent.js:383]: What do you think about accepting a string or regex? We may be able to reuse code from `allowedParentDomains`: https://github.com/search?q=repo%3Akrakenjs%2Fzoid%20allowedParentDomains&type=code

Comment (mnicpt) [reply src/parent/parent.js:383]: @gregjopa I initially was going down this path until I realized using a string won't play nice when going from environment to environment like Regex does. We could always have it point to an enum. 🤔

Comment (mnicpt) [reply src/parent/parent.js:383]: The `domain` prop does the same thing as `allowedParentDomain` in that it uses the same type def as `DomainMatcher` from `cross-domain-utils`. I just recently updated `DomainMatcher` to except an array of string or regex.

Comment (mnicpt) [reply src/parent/parent.js:383]: @gregjopa Yeah, I can do this. I'm using `matchDomain` which allows for all.

Comment (gregjopa) [reply src/parent/parent.js:383]: >@gregjopa I initially was going down this path until I realized using a string won't play nice when going from environment to environment like Regex does. 

Ahh I see. I'm glad you already considered this.

**Pull request merged**